service/src/main/java/uk/gov/hmcts/reform/fpl/service/CourtService.java

```diff
@@ -36,6 +40,19 @@ public String getCourtEmail(CaseData caseData) {
             .orElse(null);
     }
 
+    public String getPreviousCourtName(CaseData caseData) {
+        if (!caseData.getPastCourtList().isEmpty()) {
```

Comment:
I suppose this can be replaced by 

Optional.ofNullable(caseData.getPastCourtList())
                .orElse(Collections.emptyList()).stream()
                .map(Element::getValue)
                .max(comparing(Court::getDateTransferred, nullsFirst(naturalOrder())).reversed())
                .map(Court::getName)
                .orElse(null);